components/Home/Home.vue

```diff
@@ -300,14 +343,52 @@ async function fetchFavorites(ids: number[]) {
 }
 
 async function handleFavorites() {
-  return await fetchFavorites(favoritesIds.value)
-    .then((f) => {
-      favorites.value = f
-      return favorites.value
-    })
-    .catch((e) => {
-      console.error('Vido error:', (e as Error).message)
-    })
+  favoriteFeatures.value = []
+
+  const favorites = await fetchFavorites()
+  const favoriteAddresses = await fetchFavoriteAddresses()
+
+  favoriteFeatures.value = [...favoriteFeatures.value, ...favorites, ...favoriteAddresses]
+}
+
+async function fetchFavoriteAddresses() {
+  if (!favoriteAddresses.value.size)
+    return []
+
+  const promises = Array.from(favoriteAddresses.value.values()).map(async (hash) => {
+    const coordinates = decodeBase32(hash)
+    return await useFetch<ApiSearchResult<ApiAddrSearchResult>>(`${config!.API_ADDR}/reverse?lon=${coordinates.lng}&lat=${coordinates.lat}`)
+  })
+
+  const responses = await Promise.all(promises)
+  // TODO: Handle error or empty data
+  return responses
```

Comment:
Duplicate code. Same code on mount ?

Reply:
This part is duplicated ... I could make a function of it eventually.

```js
const coordinates = decodeBase32(hash)
return await useFetch<ApiSearchResult<ApiAddrSearchResult>>(`${config!.API_ADDR}/reverse?lon=${coordinates.lng}&lat=${coordinates.lat}`)
```